Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -613,8 +623,21 @@ extension PathHierarchy.DisambiguationContainer {
     /// Combines the data from this tree with another tree to form a new, merged disambiguation tree.
     func merge(with other: Self) -> Self {
         var newStorage = storage
-        for element in other.storage where !storage.contains(where: { $0.matches(kind: element.kind, hash: element.hash )}) {
-            newStorage.append(element)
+        for element in other.storage {
+            if let existingIndex = storage.firstIndex(where: { $0.matches(kind: element.kind, hash: element.hash )}) {
+                // If the same element exist in both containers, prefer to keep the Swift counterpart of it.
+                let existing = storage[existingIndex]
+                if existing.node.counterpart === element.node,
+                   element.node.symbol?.identifier.interfaceLanguage == "swift"
+                {
+                    // The "other" element is the Swift counterpart. Replace the existing element with it.
+                    newStorage[existingIndex] = element
+                } else {
+                    // The existing element is the Swift counterpart. Keep it without adding the counterpart element.
```

Comment:
nit: not a fan of the empty else block. I get that it's for the comment, but I feel like it's implied by the one on line 628 so we are "picking" the best fit.

Reply:
I updated the comment on line 628 so that I didn't feel like this else comment was needed anymore